### PR TITLE
Set node-labels in kubeletArguments

### PR DESCRIPTION
**Pull request by tbielawa**
This takes `openshift_node_labels` from your inventory and adds them to the `node-config.yaml` file as array items under the `kubeletArguments.node-labels` subkey.

Given an inventory with a host like this:

`hostname .... openshift_node_labels="{'region': 'infra', 'foo': 'bar'}" .....`

This will produce output in your `node-config.yaml` file as such:

```yaml
kubeletArguments:
  node-labels:
  - region=infra
  - foo=bar
```

**CURRENT BUG**
* If `openshift_node_labels` changes (items added/removed) only **additions** are reflected in the `node-config.yaml` file
* Removing labels from the inventory does not result in the labels being removed from the `node-config.yaml` file

Feature requests comes from the Ops team

* https://bugzilla.redhat.com/show_bug.cgi?id=1359848

### CODE REVIEW

**Comment by tbielawa** [comment]
Bugs possibly manifesting due to cached facts on the target hosts in `/etc/ansible/facts.d/openshift.fact`.

```json
[root@n01 facts.d]# less openshift.fact
{"node": {"schedulable": true, "labels": {"region": "infra", "ho": "dor"}, ... }}
```

Changing `openshift_node_labels` in your inventory to remove the `'ho': 'dor'` key will not be reflected in the `node-config.yaml`. Adding another key will be reflected.

**Comment by tbielawa** [comment]
One last note I just observed, **changing** a label value **is** reflected in the `node-config.yaml` file.

**Comment by tbielawa** [comment]
The commit added by @abutcher fixes the issue with the `node-config.yaml` file not being updated. However we've discovered a new problem.

Even with the `--overwrite` option to to the labeling command, labels saved in the nodes object (i.e. `oc get node/foo.example.com`) will not be removed. Only added/updated. The node object also stores kube specific labels so we can no just erase them all.

Erasing a label is done by giving the name of the label to remove and adding a `-` to the end of it `kubectl label pods foo bar-` would remove `bar`.

To utilize this functionality we would effectively need to calculate a diff of the current vs. expected new labels and then conditionally run the label remove command.

**Comment by abutcher** [comment]
This tracks current behavior of not removing labels but we will always set the correct labels in node `kubeletArguments` and in cached facts. Adding (optional?) support for label removal can be done in a follow-up PR, imo.

**Comment by abutcher** [comment]
aos-ci-test

**Comment by tbielawa** [comment]
aos-ci-test

**Comment by abutcher** [comment]
@detiber PTAL

**Comment by tbielawa** [comment]
@detiber @sdodson review/merge plz?

**Comment by detiber** [comment]
@wshearn @dak1n1 ptal

File: roles/openshift_node/tasks/main.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@
     role: "{{ item.role }}"
     local_facts: "{{ item.local_facts }}"
   with_items:
+  # Reset node labels to an empty dictionary.
+  - role: node
+    local_facts:
+      labels: {}
```
Comment (sdodson) [inline roles/openshift_node/tasks/main.yml]: Why was this necessary?

Comment (sdodson) [inline roles/openshift_node/tasks/main.yml]: Nevermind, it's necessary to pickup deletions.
